server/src/main/java/com/cloud/resource/ResourceManagerImpl.java

```diff
@@ -2905,16 +2905,16 @@ private boolean doCancelMaintenance(final long hostId) {
     }
 
     /**
-     * Handle agent (if available) if its not connected before cancelling maintenance.
      * Agent must be connected before cancelling maintenance.
-     * If the host status is not Up:
+     * Connect and restart agent (if available) if its not connected before cancelling maintenance.
+     * Restart agent even if it is connected to refresh the storage pools
```

Comment:
@abh1sar check if any doc update required for this behavior change during cancel maintenance.

Reply:
@sureshanaparti I didn't find anywhere in the docs that mentions anything regarding this